epaint/src/shape.rs

```diff
@@ -49,6 +49,15 @@ pub enum Shape {
         /// If true, tilt the letters for an ugly italics effect
         fake_italics: bool,
     },
+    MulticolorText {
+        /// Top left corner of the first character.
+        pos: Pos2,
+        /// The layed out text
+        galley: Galley,
+	text_style: TextStyle, // TODO: Font?
+	color_map: TextColorMap,
```

Comment:
No need to add a separate `MulticolorText` - just add `color_map: TextColorMap` to `Text`, and set it to `TextColorMap::default()` when there are no color changes